src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1074,35 +1076,30 @@ static bool processLocation
   {
     const ContextAttribute* caP = caV[ix];
 
-    for (unsigned jx = 0; jx < caP->metadataVector.size(); ++jx)
```

Comment:
I looked at the three "difficult-diff" functions and all ok. Code that is ease to follow.
However, there are two cases where you could increase readibility.
Case 1: do a 'continue' instead of a big if:
processLocation:

     if (location.length() > 0)
    {
    }

better:

    if (location.length() > 0)
    {
        continue;
    }

Then in  appendContextAttributeItem the entire function is inside a "if (legalUsage())" (or something similar).
If could really gain on starting with  "if (!legalUsage()) { ...   return }" instead.